packages/execution-environments/src/common/BaseControllers.ts

```diff
@@ -0,0 +1,250 @@
+import { Duplex } from 'stream';
+import { MetaMaskInpageProvider } from '@metamask/inpage-provider';
+import { SnapProvider } from '@metamask/snap-types';
+import { errorCodes, ethErrors, serializeError } from 'eth-rpc-errors';
+/* eslint-disable-next-line import/no-unassigned-import */
+import 'ses';
+import EEOpenRPCDocument from '../openrpc.json';
+import {
+  Endowments,
+  JSONRPCID,
+  JsonRpcRequest,
+} from '../__GENERATED__/openrpc';
+import { isJsonRpcRequest } from '../__GENERATED__/openrpc.guard';
+import { rpcMethods, RpcMethodsMapping } from './rpcMethods';
+import { sortParamKeys } from './sortParams';
+import { createTimeout } from './timeout';
+
+type SnapRpcHandler = (
+  origin: string,
+  request: JsonRpcRequest,
+) => Promise<unknown>;
+
+const fallbackError = {
+  code: errorCodes.rpc.internal,
+  message: 'Execution Environment Error',
+};
+
+export class BaseController {
```

Comment:
In line with the suggested renaming of this package:

```suggestion
export class BaseSnapExecutor {
```